### PR TITLE
Sequence behavior

### PR DESCRIPTION
**Pull request by glyph**
This is a part of the fix for #204, because it doesn't provide the specified `repeat` option; it just always repeats. I wonder if we should do it that way; perhaps we should have different names for the different things you do when you get to the end?

### CODE REVIEW

File: mimic/model/nova_objects.py [hunk 1]
```diff
@@ -527,6 +527,35 @@ def fail_later(server):
     return fail_later
 
 
+@server_creation.declare_behavior_creator("sequence")
+def sequence(parameters):
+    """
+    Sometimes a sequence of behaviors occur when you try to create a server in
+    a predictable pattern.
+
+    Takes one parameter, ``behaviors``, which is a list of specifications of
```
Comment (cyli) [inline mimic/model/nova_objects.py]: Would it make sense to document here that like the list is a dictionary containing the name under the key 'name' and the parameters under the name 'parameters'?  This specification is different than the other behavior specifications (since they just take a dictionary as parameters?)

File: mimic/model/nova_objects.py [hunk 2]
```diff
@@ -527,6 +527,35 @@ def fail_later(server):
...
+            server_creation.create_behavior(behavior["name"],
+                                            behavior["parameters"])
+            if behavior["name"] != "default"
+            else server_creation.default_behavior
+        )
+        for behavior in behavior_specification
+    ]
+
+    def rotating_behavior(collection, http, json, absolutize_url):
+        current = behavior_objects[rotating_behavior.index]
```
Comment (cyli) [inline mimic/model/nova_objects.py]: Maybe `itertools.repeat` can be used here?

File: mimic/model/nova_objects.py (hunk 1 above)
Comment (glyph) [inline mimic/model/nova_objects.py]: This … isn't quite true? Normally they take a name under "name" and parameters under "parameters" and criteria under "criteria"; here, the criteria are already specified.  Perhaps an example dictionary would be helpful though.

File: mimic/model/nova_objects.py (hunk 2 above)
Comment (glyph) [inline mimic/model/nova_objects.py]: I think you mean `itertools.cycle`?  I explicitly wanted to do repeating in a separate pull request, though, as the description said.

File: mimic/model/nova_objects.py (hunk 1 above)
Comment (cyli) [inline mimic/model/nova_objects.py]: Sorry, I mean they they are documented with name and parameters in a dictionary in the REST api, but not in the object model. The behaviors above describes the dictionary that is in parameters.

Whereas the behavior specification here is expecting a list of dictionaries as would be passed as the body to the REST api call.

Comment (glyph) [inline mimic/model/nova_objects.py]: Fundamentally you're right, this documentation is indecipherable, I will include an example and describe it more clearly.

**Comment by glyph** [comment]
Hmm… I was thinking "with minor changes" and "if tests pass" would be mutually exclusive – I really wish github let you put a description on labels so we could explain them in more detail :-\

**Comment by glyph** [comment]
In other words, "with minor changes" would mean "change it and then land", but "if tests pass" would mean "land exactly as-is as long as the tests pass".

**Comment by glyph** [comment]
In both cases you'd need to wait for a passing test run.

**Comment by cyli** [comment]
@glyph - I figured they were composable - e.g. if tests pass after minor changes.

**Comment by glyph** [comment]
Does the docstring look good now though? This is kind of subtle, and I think we don't have the greatest docs, so I'd really like an affirmative "yes" before pulling the trigger on this :).

**Comment by cyli** [comment]
@glyph +1

**Comment by glyph** [comment]
Given the build times on travis, I'm going to optimistically assume that the previous correct build results and the smattering of test passes and coverage results on the current code are good enough for a merge.
